### ISSUE TITLE
electron-log when using nodeIntegration: false and preload script

### ISSUE THREAD

**Issue by avner-hoffmann** [megahertz/electron-log#126]
electron-log version : 3.0.3 (upgraded from 2.2.17)

I'm creating a browser window using nodeIntegration: false and using preload script to create an API between the main and renderer processes. 
Since I can't require electron-log from the renderer process, I require the module in the preload script and pass it to the API as follows:

```
// preload.js

const electronLog = require('electron-log'); 
window.Bridge = {
    electronLog
};

``` 
That way the log is available in the renderer process.
I've noticed that sometimes the log output in the log file from the renderer process is getting deleted by an output from the main process after getting printed, as if the file isn't open for modifications with append mode but with create/write mode so it gets deleted

Any clue why can this happen?

**Comment by megahertz** [comment]
Do you use default settings? Maybe the log is archived to log.old.log?

**Comment by avner-hoffmann** [comment]
I'm setting the log level and I'm setting the max file size to 20MB
These are the same settings I've used in version 2.2.17 and haven't seen an issue
It looks like it might be a thread synchronization issue, as if two process write to the log file in the same time, but I guess the OS should block this anyway when opening a file with Append mode

**Comment by megahertz** [comment]
ok, I'll dive into this problem later.

**Comment by avner-hoffmann** [comment]
Hi @megahertz - Is there any progress with this issue?
I really want to upgrade to version 3.x but that prevents it. Have you got the change to investigate it ?

**Comment by megahertz** [comment]
I think, I'll start working on that changes in the second half of August, but I can't promise.

**Comment by megahertz** [comment]
In v4 main and renderer processes use different files, so there is no such a problem with default configuration.

Currently, I can't found a good workaround (which is very stable and have no impact on performance) for situation where multiple processes use the same file. But I'll try to investigate this problem later.

I have a problem with reproducing such situation. Could you tell me about your environment where the problem appears? Which OS, how frequently logging is called and how big are log records?

**Comment by avner-hoffmann** [comment]
Hi @megahertz -
On v4 of electron-log the log files for the main and renderer process are separate?
Is this configurable? Personally our application needs the log centralized in one file. I think it's necessary to allow this since a flow can start on one process and continue to another and it would be very difficult to track if each process writes its logs to a separate file 

Regarding your questions, I recall testing it on both macOS - High Sierra and Mojave and Windows 10.
Logging is called very frequent, in can be even 10 of lines in a second. We configure the log files to be up to 20MB before log rotation

**Comment by megahertz** [comment]
@avner-hoffmann Thank you for feedback. 

In the new version you can set a single log file as it was before.

> Logging is called very frequent, in can be even 10 of lines in a second. 

It's not so many. Maybe you log some big objects? HDD or SSD?

**Comment by megahertz** [comment]
I've played a lot with concurrent log writing. The main problem for me - I've never got a situation when one process overrides data written by another process. Even if two processes write 2gb data in 10 seconds - there are no corrupted records. Even if there are a lot of attempts.

So, I close this issues while there is no such a problem with default configuration. 

But I would be very grateful if someone helps me to reproduce the problem. I've created [a simple app](https://github.com/megahertz/electron-log-experiments/tree/concurrent-write) which writes ~1kb of data 10000 times (by default) from two processes at the same time. If log file is corrupted than a CI build fails.
